### DISCUSSION TITLE
请问用chromego成功后，如何共享给obs，实现油管直播？

### DISCUSSION THREAD

**Discussion by DannyJRu** [bannedbook/fanqiang#689] [Q&A] [answered]
我用的是“1.clashB翻墙”成功的，但是obs推流给油管时总是连不上

**Comment by bannedbook** [comment, marked as answer]
简单的说你在中国大陆的话，网络原因没法直接连接twitch和youtube或者其他类似的直播网站，你可以在OBS log里看错误信息。而且像OBS这种直播软件没有内置上级代理功能，毕竟不是主流需求，所以需要有手动把OBS指向一个proxy。

有几种比较主流的办法。
1.开启一个全局proxy，代理本机全部请求。
2.在本机使用软件强制让OBS指向一个proxy。
3.在路由上开启一个proxy，让它对本机提供透明代理。

第一种存在比较多的问题，不太适合直播。如果你的proxy软件本身支持网卡劫持并且分流的办法，那也是个不错选择。

第二种之前的帖子里我提了proxifier和SocksCap64。但是当时是2017年，过了这么多年直播技术发展也挺快。proxifier对现在常用的rtmp直播协议已经存在兼容问题了，表现是断流或者干脆无法响应，我猜测是proxifier没有提供udp完整支持的缘故，导致会误判服务器断开连接。
所以我建议使用SocksCap64，作者的最后一个版本是4.7， 然后因为某种原因数据丢失了 。
[https://drive.google.com/open?id=1x5naJeK2YkV6QCYUlUg5QNMl1Izf4-ti]


比较遗憾的就是只有windows平台，而且不会更新了。

还有一款还在更新的是Netch， 功能类似SocksCap64+SStap，可以选择自己需要的模式，本身也有封装了V2ray可以直接作为proxy软件使用。但是现在也是不兼容OBS。
[https://github.com/netchx/Netch/releases]
其他平台我没有测试过，mac或者Linux可以试试proxychains或者tsocks。

第三种可以去openwrt/lede之类的专业论坛了解。简单地说你在路由上将推流服务器域名直接指向proxy，对本机来说就没有网络问题了。
注意下你的proxy是否有使用加密，一般是AES之类的，cpu建议使用支持aes加密指令集的，更稳定一些。看支不支持aes等可以把你的CPU型号复制到intel ark查询，不建议使用arm的，同价位性能弱的多。

对于观众来说，一般YouTube的域名常用PAC中是内置的，Twitch可能没有。
chrome可以安装一个Proxy SwitchyOmega的扩展。
具体教程Google一下也挺详细的，类似如下添加:


最后想说下OBS现在已经支持win10的chrome窗口捕获了。
以前很多人发现win10下OBS截取chrome或者poi之类的工具是个黑屏，这个是OBS捕获原理决定的，抓不到内容。需要改成win7兼容启动，现在不需要了。
在最新版的OBS里重新捕获一次窗口，OBS会自动使用win10的新方式捕获chrome或者poi。

**Reply by DannyJRu** [reply]
非常感谢，你说的很多我都不太懂，但是我在油管上看到有用proxifier实现的教程
